Automattic/wp-calypso
# Components: Wire up reduxified country states

Moves away from `lib/states-list` to using `countryStates` built on Redux.

To test:
1. Go to `My Sites`.
1. Under `Domains`, click `Add`.
1. Hit `Select` for any domain to enter the checkout process.
1. Skip adding Email support.
1. In the `Domain Contact Information` form, select `United States` in the country dropdown. The `State` input field should have been replaced with a dropdown of states.

See #8202.
Fixes #7922.
